src/main/java/com/thebookcooper/controller/BookTransactionController.java

```diff
@@ -0,0 +1,126 @@
+            }
+            return new ResponseEntity<>("No transactions found", HttpStatus.NOT_FOUND);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>("Error retrieving transaction count", HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @PostMapping("/create")
+    public ResponseEntity<?> createTransaction(@RequestBody String json) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            Map<String, Object> inputMap = objectMapper.readValue(json, new TypeReference<Map<String, Object>>() {});
+            BookTransaction newTransaction = new BookTransaction();
+            Connection connection = dcm.getConnection();
+            BookTransactionDAO transactionDAO = new BookTransactionDAO(connection);
+
+            newTransaction.setBuyerId(Long.parseLong(inputMap.get("buyerId").toString()));
+            newTransaction.setSellerId(Long.parseLong(inputMap.get("sellerId").toString()));
+            newTransaction.setListingId(Long.parseLong(inputMap.get("listingId").toString()));
+            newTransaction.setTransactionPrice(Double.parseDouble(inputMap.get("transactionPrice").toString()));
+            newTransaction.setTransactionDate(new java.sql.Timestamp(System.currentTimeMillis()));
+            newTransaction.setTransactionStatus((String) inputMap.get("transactionStatus"));
+
+            BookTransaction createdTransaction = transactionDAO.create(newTransaction);
+            return new ResponseEntity<>(createdTransaction, HttpStatus.CREATED);
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>("Invalid JSON input", HttpStatus.BAD_REQUEST);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>("Failed to create the transaction", HttpStatus.INTERNAL_SERVER_ERROR);
+        } catch (NumberFormatException e) {
+            return new ResponseEntity<>("Invalid number format", HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping("/update/{id}")
+    public ResponseEntity<?> updateTransaction(@PathVariable("id") long id, @RequestBody String json) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            Map<String, Object> inputMap = objectMapper.readValue(json, new TypeReference<Map<String, Object>>() {});
+            Connection connection = dcm.getConnection();
+            BookTransactionDAO transactionDAO = new BookTransactionDAO(connection);
+
+            BookTransaction updatedTransaction = new BookTransaction();
+            updatedTransaction.setTransactionId(id);
+            updatedTransaction.setBuyerId(Long.parseLong(inputMap.get("buyerId").toString()));
+            updatedTransaction.setSellerId(Long.parseLong(inputMap.get("sellerId").toString()));
+            updatedTransaction.setListingId(Long.parseLong(inputMap.get("listingId").toString()));
```

Comment:
here again with the transaction date update-- should it also be update here?

Reply:
I just realized i made the same comment on the same line but it was supposed to be in the point transaction file